salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,200 @@
+                    missing_deps
+                ))
+            )
+
+    # move this to yaml file
+    @parameterized.expand([
+        (None, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {}, ('kubeconfig.conf', 'mycontext')),
+        (None, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        (None, {"kubeconfig": "kubeconfig.conf"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        (None, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, None, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, None, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"context": "mycontext"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
```

Comment:
Please only provide `metalk8s.api_server` content in the arguments and put it in the right place in the pillar directly in the test function